### ISSUE TITLE
about re-training (Transfer Learning) with quantize aware training

### ISSUE THREAD

**Issue by sinatv52** [soon-yau/QNN#3]
Hi, I want to train a custom image classifier and implement it in FPGA using HLS (inference only). currently i am using google colab for training. 
therefore i cant train the network from scratch. it is really time consuming.
is it possible to use quantize aware training and transfer learning or fine tuning? (i think so)
do you have any example, tutorial or recommendations? or can you tell me the procedure?
Thanks. This tutorial is really awesome.

**Comment by soon-yau** [comment]
In I/O 19, Google says that post training quantization achieved accuracy that's very close (<1% difference) to quantization aware training. Therefore you should probably try that first. 

If you have both the training code and saved model (not the frozen graph), you can add tf.contrib.quantize.create_training_graph() (see tutorial 3) after the saved graph has been loaded to generate a quantization aware graph and start training from there. But please make a backup of your code and graph in case anything goes wrong.

**Comment by sinatv52** [comment]
sorry for commenting on a closed issue. i know this is not an issue related to your great tutorials. i have searched forums and websites like stackoverflow or medium but i could not find the answer.

as you recommended, i watched the i/o 19 and found the docs about post quantization and tried my custom model and the results were great and rather no drop in the accuracy. 
but because i am doing a research about quantization and implementation on FPGA, i want to do sth like what you did. training and evaluating quantized model with different bit width.

the problem i have is that with a pretrained model i want to all low level quantization api. and still use pretrained values in the saved checkpoints. i searched and found [this  ](https://medium.com/@chenchoulo/tensorflow-remove-nodes-add-nodes-with-pre-trained-model-4fb8cf44b531) but it seems confusing and not working. do you have any reference for adding new nodes and still using pretrained checkpoints?
